File: yy_videolib/src/main/java/com/ybj366533/videolib/utils/LogUtils.java
package com.ybj366533.videolib.utils;


import android.util.Log;



/**
 * 日志管理
 */

public class LogUtils   {

    private static final String TAG = "YYVideo";

    public static final int DEBUG = 3;
    public static final int INFO = 4;
    public static final int WARN = 5;
    public static final int ERROR = 6;

    private static int level = DEBUG;
    public static void setLogLevel(int logLevel) {
        level = logLevel;
    }

    private static boolean sIsLoggable = true;

    public static void setIsLoggable(boolean isLoggable) {
        sIsLoggable = isLoggable;
    }


    public static void LOGS(String message) {
        if (message == null || sIsLoggable == false) return;
        Log.e(TAG, message);
    }

    public static void LOGD(String tag, String message){
        if (message == null || sIsLoggable == false || level > DEBUG) return;
        Log.d(TAG + "-" +tag, message);
    }

    public static void LOGI(String tag, String message){
        if (message == null || sIsLoggable == false || level > INFO) return;
        Log.w(TAG + "-" +tag, message);
    }

    public static void LOGW(String tag, String message){
        if (message == null || sIsLoggable == false || level > WARN) return;
        Log.w(TAG + "-" +tag, message);
    }

    public static void LOGE(String tag, String message){
        if (message == null || sIsLoggable == false || level > ERROR) return;
        Log.e(TAG + "-" +tag, message);
    }

    public static void DebugLog(String tag, String msg) {
        //Log.e(tag, msg);
        Log.i(TAG + "-" +tag, "[DebugLog] will be deleted, please use LOGI etc. " +msg);
    }

}

File: app/src/main/kotlin/com/ybj366533/yycamera/base/utils/LogUtils.kt
package com.ybj366533.yycamera.base.utils

import android.util.Log
import com.ybj366533.yycamera.BuildConfig

/**
 * Log封装
 * Created by summer on 2017/10/30.
 */

enum class LogLevel {
    Verbose, Debug, Info, Warn, Error, WTF, Assert
}

var TAG = "yy"
var space = "->"
var isDebug = BuildConfig.DEBUG

val trace get() = trace(4)

private fun trace(index: Int): String {
    val caller = Throwable().stackTrace[index]

    // 1. 类名 + 方法名
    //val callerClazzName = caller.className
    //val callerClazzName = callerClazzName = callerClazzName.substring(callerClazzName.lastIndexOf(".") + 1);
    //var template = "${callerClazzName}.${caller.methodName}(${caller.fileName}:${caller.lineNumber})"

    // 2. 方法名

    var template = "${caller.methodName}(${caller.fileName}:${caller.lineNumber})"
    return template
}

fun log(msg: String, level: LogLevel = LogLevel.Error, tag: String = TAG) = run {
    if (isDebug) {
        when (level) {
            LogLevel.Verbose -> {
                Log.v(tag, "$trace $space $msg")
            }
            LogLevel.Debug -> {
                Log.d(tag, "$trace $space $msg")
            }
            LogLevel.Info -> {
                Log.i(tag, "$trace $space $msg")
            }
            LogLevel.Warn -> {
                Log.w(tag, "$trace $space $msg")
            }
            LogLevel.Error -> {
                Log.e(tag, "$trace $space $msg")
            }
            LogLevel.WTF -> {
                Log.wtf(tag, "$trace $space $msg")
            }
            else -> {
            }
        }
    }
}

fun log(msg: String, tr: Throwable, level: LogLevel = LogLevel.Error, tag: String = TAG) = run {
    if (isDebug) {
        when (level) {
            LogLevel.Verbose -> {
                Log.v(tag, "$trace $space $msg", tr)
            }
            LogLevel.Debug -> {
                Log.d(tag, "$trace $space $msg", tr)
            }
            LogLevel.Info -> {
                Log.i(tag, "$trace $space $msg", tr)
            }
            LogLevel.Warn -> {
                Log.w(tag, "$trace $space $msg", tr)
            }
            LogLevel.Error -> {
                Log.e(tag, "$trace $space $msg", tr)
            }
            LogLevel.WTF -> {
                Log.wtf(tag, "$trace $space $msg", tr)
            }
            else -> {
            }
        }
    }
}